components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -93,46 +111,66 @@ private AuthenticatorAdapterConstants.UserType resolveIdpType() {
                 AuthenticatorAdapterConstants.UserType.LOCAL : AuthenticatorAdapterConstants.UserType.FEDERATED;
     }
 
-    private User resolveLocalUserFromUserStore(AuthenticationContext context, AuthenticatedUserData.UserStore userStore)
+    private String resolveUserId() throws ActionExecutionResponseProcessorException {
+
+        if (StringUtils.isNotBlank(user.getUser().getId())) {
+            return user.getUser().getId();
+        }
+        throw new ActionExecutionResponseProcessorException("User Id is not found in the authenticated user data.");
```

Comment:
requires diagnostic

Reply:
Tracking with https://github.com/wso2/product-is/issues/22753